lib/Basics/process-utils.cpp

```diff
@@ -213,6 +213,38 @@ ExternalProcess* TRI_LookupSpawnedProcess(TRI_pid_t pid) {
   return nullptr;
 }
 
+std::optional<ExternalProcessStatus> TRI_LookupSpawnedProcessStatus(
+    TRI_pid_t pid) {
+  {
+    MUTEX_LOCKER(mutexLocker, ExternalProcessesLock);
+    auto found = std::find_if(
+        ExternalProcesses.begin(), ExternalProcesses.end(),
+        [pid](ExternalProcess const* m) -> bool { return m->_pid == pid; });
+    if (found != ExternalProcesses.end()) {
+      ExternalProcessStatus ret;
+      ret._status = (*found)->_status;
+      ret._exitStatus = (*found)->_exitStatus;
+      return std::optional<ExternalProcessStatus>{ret};
+    }
+  }
+  return std::nullopt;
+}
+
+#ifdef _WIN32
+HANDLE TRI_getProcessHandle(pid) {
+  {
+    MUTEX_LOCKER(mutexLocker, ExternalProcessesLock);
+    auto found = std::find_if(
+        ExternalProcesses.begin(), ExternalProcesses.end(),
+        [pid](ExternalProcess const* m) -> bool { return m->_pid == pid; });
+    if (found != ExternalProcesses.end()) {
+      return (*found)->_process;
+    }
+  }
+  return INVALID_HANDLE_VALUE;
+}
+#endif
```

Comment:
If this function is not going to be exposed via the header file (and it doesn't look like it is), then it should not be named `TRI_...`. Instead, it should be placed in the anonymous namespace at the top of the file.